packages/create-plugin/src/utils/utils.git.ts

```diff
@@ -0,0 +1,37 @@
+import { exec as nodeExec } from 'node:child_process';
+import { promisify } from 'node:util';
+
+const exec = promisify(nodeExec);
+
+export async function isGitDirectory() {
+  let command = 'git rev-parse --is-inside-work-tree';
+
+  try {
+    const response = await exec(command);
+
+    return response.stdout.trim() === 'true';
+  } catch (error) {
+    if (error instanceof Error && error.message.includes('fatal: not a git repository')) {
+      return false;
+    }
+
+    throw new Error(
+      `Failed to detect if you are in a git directory (used "${command}"). 
+This check is necessary so we don't accidentally override any changes you might have made to your files. 
+If you are certain you would like to continue, you can run the command again with the --force flag.`
+    );
+  }
+}
+
+export async function isGitDirectoryClean() {
+  try {
+    let command = 'git status --porcelain';
+    const response = await exec(command);
+
+    return response.stdout.trim() === '';
+  } catch (error) {
+    throw new Error(`Failed to detect if your git directory is clean (used "git status --porcelain").
```

Comment:
Similar to the previous comment. By throwing our own error with our own message we are hiding the real error behind git. We should throw the original error if we are going to throw an error here and exit.

Reply:
Fixed in https://github.com/grafana/plugin-tools/pull/702/commits/907a1e2123438cab41adf6108a82d9d853640fe7.